src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+    m_billTableWidget->setColumnCount(4);
+
+    // Set the table widget headers to the appropriate fields each Bill displays
+    m_billTableWidget->setHorizontalHeaderLabels(QString(m_BILL_NAME_COLUMN_HEADER_STRING + ";" + m_BILL_AMOUNT_DUE_COLUMN_HEADER_STRING + ";" + m_BILL_DUE_DATE_COLUMN_HEADER_STRING + ";" + m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING).split(";"));
+
+    // Initialize the row we're setting to zero
+    int row = 0;
+
+    // Create an iterator to iterate over the bill map
+    QMap<QString, Bill>::iterator billMapIterator;
+
+    for(billMapIterator = m_billMap.begin(); billMapIterator != m_billMap.end(); ++billMapIterator)
+    {
+        // Retrieve the current Bill being checked
+        Bill currentBill = *billMapIterator;
+
+        // Create a date edit for displaying and allowing the editing of the bill's due date
+        QDateEdit *dateEdit = new QDateEdit(this);
+
+        // Allow the display of a calendar for editing purposes
+        dateEdit->setCalendarPopup(true);
+
+        // Set the date edit based on the bill's due date
+        dateEdit->setDate(currentBill.getDueDate());
+
+        QComboBox *fundedStatusBox = new QComboBox(this);
+        fundedStatusBox->addItem(m_NOT_FUNDED_STRING);
+        fundedStatusBox->addItem(m_FUNDED_STRING);
+        currentBill.isFunded() ? fundedStatusBox->setCurrentIndex(1) : fundedStatusBox->setCurrentIndex(0);
+
+        // Set the columns appropriately to the Bill's attributes
+        m_billTableWidget->setItem(row, 0, new QTableWidgetItem(currentBill.getName()));
+        m_billTableWidget->setItem(row, 1, new QTableWidgetItem(QString::number(currentBill.getAmountDue(), 'f', 2)));
+        m_billTableWidget->setCellWidget(row, 2, dateEdit);
+        m_billTableWidget->setCellWidget(row, 3, fundedStatusBox);
+
+        // Increment the row for the next Bill
+        row++;
+    }
+}
+
+QDate MainWindow::convertDateStringToDate(QString p_dateString)
+{
+    // Convert the due date string into a date object and return it
+    return QDate::fromString(removeSpaces(p_dateString),"M/d/yyyy");
+}
+
+void MainWindow::readConfigAndCreateUI()
+{
+    // Attempt to open the pre-existing config file
+    QSettings m_settings(m_CONFIG_FILE_DIRECTORY, QSettings::IniFormat);
+
+    // If attempting to open the config file resulted in an error
+    if(m_settings.status() != QSettings::NoError)
+    {
+        // Alert the user with an appropriate message box
+        createCorruptConfigFileBox();
+    }
+
+    // If the config file was opened successfully
+    else
+    {
+        // Store all the keys from the config file in a list
+        QStringList configFileKeys = m_settings.allKeys();
+
+        // Iterate over each key
+        foreach(QString key, configFileKeys)
+        {
+            // The keys are stored in "Group label/key" format so they need to be separated
+            QStringList keyValues = m_settings.value(key).toStringList();
```

Comment:
Since each key has only value in this case, do we need to store the value as a QStringList? Can it just be a QString?

Reply:
Yes, changed to a String as opposed to QStringList.